node/sc/bridge_addr_journal.go

```diff
@@ -24,37 +24,51 @@ import (
 	"errors"
 	"io"
 	"os"
+	"strings"
+	"sync"
 
 	"github.com/klaytn/klaytn/common"
 	"github.com/klaytn/klaytn/node/sc/bridgepool"
 	"github.com/klaytn/klaytn/rlp"
 )
 
 var (
-	errNoActiveAddressJournal = errors.New("no active address journal")
-	errDuplicatedJournal      = errors.New("duplicated journal is inserted")
-	errEmptyBridgeAddress     = errors.New("empty bridge address is not allowed")
+	ErrNoActiveAddressJournal = errors.New("no active address journal")
+	ErrDuplicatedJournal      = errors.New("duplicated journal is inserted")
+	ErrDuplicatedAlias        = errors.New("duplicated alias")
+	ErrEmptyBridgeAddress     = errors.New("empty bridge address is not allowed")
+	ErrEmptyJournalCache      = errors.New("empty bridge journal")
+	ErrEmptyBridgeAlias       = errors.New("empty bridge Alias")
+	ErrNotAllowedAliasFormat  = errors.New("Not allowed bridge alias format")
 )
 
 // bridgeAddrJournal is a rotating log of addresses with the aim of storing locally
 // created addresses to allow deployed bridge contracts to survive node restarts.
 type bridgeAddrJournal struct {
-	path   string         // Filesystem path to store the addresses at
-	writer io.WriteCloser // Output stream to write new addresses into
-	cache  map[common.Address]*BridgeJournal
+	path       string         // Filesystem path to store the addresses at
+	writer     io.WriteCloser // Output stream to write new addresses into
+	cache      map[common.Address]*BridgeJournal
+	aliasCache map[string]common.Address
+	writerMu   *sync.Mutex
```

Comment:
`Do we always write to BridgeAddrJournal?` I'm sorry, I couldn't get correct understanding. Could you elaborate on it?
` Or just reading the journal is rare so using RWMutex instead won't make any difference?` Do you mean RWMutex is not necessary?
The number of access of `cache` and `aliasCache` depends on what APIs are called.

Reply:
I was wondering why you used Mutex for `BridgeAddrJournal` and RWMutex for `cache`. You used RWMutex for `cache` so that concurrent reading would be allowed. So if reading the `BridgeAddrJournal` is frequent, we could use RWMutex too, for efficiency. But still, you used Mutex for `BridgeAddrJournal`, and I was curious about the reason.

I have no comment for using RWMutex for `cache`, because it seems quite reasonable.

Reply:
Got it. The `writerMu` is a guard for the `journal.writer` variable. All the related operation with this variables are file write operations. Thus, `RWMutex` was not chosen. Could you review them again?

Reply:
If I understand right, `load` is reading from the journal. But I found out that it is executed once a node starts, so using RWMutex won't be needed.

Thanks!

Reply:
> If I understand right, load is reading from the journal. But I found out that it is executed once a node starts.

Exactly.